common/djangoapps/third_party_auth/settings.py

```diff
@@ -40,6 +41,9 @@ def apply_settings(django_settings):
     # Adding extra key value pair in the url query string for microsoft as per request
     django_settings.SOCIAL_AUTH_AZUREAD_OAUTH2_AUTH_EXTRA_ARGUMENTS = {'msafed': 0}
 
+    # Avoid default username check to allow non-ascii characters
+    django_settings.SOCIAL_AUTH_CLEAN_USERNAMES = not settings.FEATURES.get("ENABLE_UNICODE_USERNAME")
```

Comment:
Will doing this allow special characters to make it into the username if `ENABLE_UNICODE_USERNAME` is true, or is that caught at a different layer?

Reply:
Yes that's correct. If `ENABLE_UNICODE_USERNAME` is true, then the username validation checks (by default, the python-social-auth's [`clean_username`](https://github.com/python-social-auth/social-core/blob/master/social_core/storage.py#L123) is called which basically  checks for ascii and special characters) is skipped.

Reply:
Okay. Please just add a comment that explains this to where `ENABLE_UNICODE_USERNAME` is defined in `common.py` so that people understand the implications. I'll approve once that's in.

Thank you.

Reply:
Added a comment [here](https://github.com/edx/edx-platform/blob/e8b9443bfe79dcdc3dedce99423b9da2ce65a5fc/lms/envs/common.py#L847-L848).